src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerSpec.scala

```diff
@@ -1901,6 +1902,71 @@ class JsonRpcControllerSpec
     response should haveObjectResult("transactions" -> JArray(expectedTxs.toList))
   }
 
+  "request pending transactions and return valid response" should "mempool is empty" in new TestSetup {
```

Comment:
I think we should start to split this file. It is already too big. Maybe sth like `JsonRpcControllerHandlingTransactionsSpec`? WDYT?

Reply:
Yes, I highly agree. I try to do this in this PR but work it is a lot of changes and it shadows the real work done in here + somehow tests got broken. So it is better to go small steps.

I split JsonRpc tests in separate PR: https://github.com/input-output-hk/mantis/pull/734